### PR TITLE
Gh963 copy and show envs

### PR DESCRIPTION
**Pull request by FrankMittelbach**
# Internal housekeeping

## Status of pull request

- Under development
- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

@PhelypeOleinik could you perhaps take a look at sensibly defining `\ShowEnvironment` so that it works for old style *and* new style environments?

### CODE REVIEW

File: base/doc/usrguide.tex [hunk 1]
```diff
@@ -926,6 +923,26 @@ \section{Copying and showing (robust) commands}
 \cs{NewDocumentCommand}, etc.\ it also gives you detailed information
 about the argument signature.
 
+\begin{decl}
+  |\NewEnvironmentCopy|       \arg{env} \arg{existin-env} \\
```
Comment (mrpiggi) [inline base/doc/usrguide.tex:927]: "g" missing in `existin-env`

File: base/ltdefns.dtx [hunk 2]
```diff
@@ -784,7 +784,9 @@
 %    not take arguments. (but may contain |\par|s)
 %
 % Make sure that an attempt to define a `graf' or `group' environment
-% fails.
+% fails by temporaily letting the undefined \verb=\...= (begin code) to
```
Comment (mrpiggi) [inline base/ltdefns.dtx:787]: temporarily

**Comment by PhelypeOleinik** [comment]
@FrankMittelbach I'll do that this weekend.

File: base/doc/usrguide.tex [hunk 3]
```diff
@@ -926,6 +923,26 @@ \section{Copying and showing (robust) commands}
 \cs{NewDocumentCommand}, etc.\ it also gives you detailed information
 about the argument signature.
 
+\begin{decl}
+  |\NewEnvironmentCopy|       \arg{env} \arg{existing-env} \\
+  |\RenewEnvironmentCopy|     \arg{cmd} \arg{existing-env} \\
+  |\DeclareEnvironmentCopy|   \arg{cmd} \arg{existing-env}
```
Comment (muzimuzhi) [inline base/doc/usrguide.tex:929]:
```suggestion
  |\RenewEnvironmentCopy|     \arg{env} \arg{existing-env} \\
  |\DeclareEnvironmentCopy|   \arg{env} \arg{existing-env}
```

Comment (PhelypeOleinik) [reply base/doc/usrguide.tex]: Thanks

**Comment by PhelypeOleinik** [comment]
Should be done.  I don't know what is (if there is one) right way to present an environment.  The different possibilities (that I saw) are in the final four examples in the test file `github-0963`.

**Comment by FrankMittelbach** [comment]
> Should be done. I don't know what is (if there is one) right way to present an environment. The different possibilities (that I saw) are in the final four examples in the test file `github-0963`.

I think it is fine as it is now. This is a debugging help and what is needed is getting the right definitions shown which is what happens, the precise output formatting is of less importance imo.

**Comment by PhelypeOleinik** [comment]
I fixed a couple more issues.  If tests pass, it's good to go from my end.

**Comment by FrankMittelbach** [comment]
> I fixed a couple more issues. If tests pass, it's good to go from my end.

well, not all do, but it looks easily mendable :-)

**Pull request merged**